### PR TITLE
Dirty-Model-Saves: Allows models to be updated when invalid for dirty…

### PR DESCRIPTION
**Pull request by nickolaswilson**
… saving.

### CODE REVIEW

File: dist/availity-angular-ui.js [hunk 1]
```diff
@@ -628,6 +640,15 @@
           avValField.validate(ngModel.$viewValue);
         });
 
+        // Removes all errors on page, does not reset view or model values, this is to be handled by the app
+        scope.$on(AV_VAL.EVENTS.RESET, function () {
+          var violations = ngModel.avResults.violations;
+          ngModel.avResults.violations = [];
+          element.parents(AV_BOOTSTRAP_ADAPTER.CLASSES.FORM_GROUP).removeClass('has-error');
```
Comment (robmcguinness) [inline dist/availity-angular-ui.js]: The original intent for the validation framework was to decouple the validation for a field element from the UI rendering of a validation error.  This couples them together.

**Comment by robmcguinness** [comment]
@ivern @bobbennett  what are you thoughts on allowing the validation framework to pass invalid data to the models? Seems like a strange request but maybe I'm missing the forest from the trees?  @nickolaswilson can you explain the use case?

**Comment by nickolaswilson** [comment]
The idea behind this is allowing a user to save invalid data. For example our app has a rule that you can't have a date more than a calendar years ago in a field. A user thinks they know the date and puts one in that's 2 calendar years ago. It will validate false in the UI. A customer walks up to the person entering this data and now the employee must attend the customer. Instead of taking the time to fix all the errors on the page they can just hit 'Save and Continue' and it will save all the data regardless of the validations. Now in this case we are saying 'Save and Continue' is not the same as 'Submit'. A user would not be able to submit the data unless it was all valid. We're capable of saving any entered data (as long as types match (int to int, etc; cause these are all validated in Java too) and incomplete data is not passed like half an ISO date), but we only prep data for sending when a user submits validated data.

**Comment by robmcguinness** [comment]
it sounds like Gmail's automatic draft save feature which I think is great...saving invalid models not so much.  i'll wait for the others to chime in.

**Pull request merged**